stellargraph/utils/saliency_maps/integrated_gradients.py

```diff
@@ -28,109 +28,162 @@
+        total_gradients = np.zeros(self.X.shape)
         for alpha in np.linspace(0, 1, steps):
             X_step = X_baseline + alpha * X_diff
-            total_gradients += self.get_node_masks(
-                node_idx, class_of_interest, X_step, A_index, A_val
-            )
+            if self.is_sparse:
+                grads = self._compute_gradients(
+                    [
+                        X_step,
+                        np.array([[node_idx]]),
+                        self.A_indices,
+                        self.A,
+                        class_of_interest,
+                    ],
+                    variable="nodes",
+                )
+            else:
+                grads = self._compute_gradients(
+                    [X_step, np.array([[node_idx]]), self.A, class_of_interest],
+                    variable="nodes",
+                )
+
+            total_gradients += grads
+
         return np.squeeze(total_gradients * X_diff, 0)
 
     def get_integrated_link_masks(
         self,
         node_idx,
         class_of_interest,
         non_exist_edge=False,
-        X_val=None,
-        A_index=None,
-        A_val=None,
-        steps=20,
         A_baseline=None,
+        steps=20,
     ):
         """
         Args:
-        X_val, A_val, node_idx, class_of_interest: The feature matrix, adjacency matrix, target node index and class of interest
-                                                   which are used to compute the vanilla gradients.
-        X_baseline: For integrated gradients, X_baseline is the reference X to start with. Generally we should set X_baseline to a all-zero
-                                              matrix with the size of the original adjacency matrix.
-        steps (int): The number of values we need to interpolate. Generally steps = 20 should give good enough results.\
-
-        non_exist_edge (bool): Setting to True allows the function to get the importance for non-exist edges. This is useful when we want to understand
-            adding which edges could change the current predictions. But the results for existing edges are not reliable. Simiarly, setting to False ((A_baseline = all zero matrix))
-            could only accurately measure the importance of existing edges.
+            node_idx: the index of the node to calculate gradients for.
+            class_of_interest: the index for the class probability that the gradients will be calculated for.
+            A_baseline: For integrated gradients, A_baseline is the reference adjacency matrix to start with. Generally
+                we should set A_baseline to an all-zero matrix or all-one matrix with the size of the original
+                A_baseline matrix.
+            steps (int): The number of values we need to interpolate. Generally steps = 20 should give good enough results.
 
         return (Numpy array): shape the same with A_val. Integrated gradients for the links.
```

Comment:
Could update this `return` docstring like you've done with the other method 😄

Reply:
missed that! yep, will do